model/CSVDialog.cc

```diff
@@ -34,6 +53,111 @@ void CSVDialog::reportFromFile(const std::string& input, const std::string& outp
   reportFromCSVFile(is,of,spec);
 }
 
+// Return file name after downloading a CSV file from the web.
+std::string CSVDialog::loadWebFile(const std::string& url)
+{
+    try
+    {
+		// Parse input URL
+        boost::regex ex("(http|https)://([^/ :]+):?([^/ ]*)(/?[^ #?]*)\\x3f?([^ #]*)#?([^ ]*)");
+        boost::cmatch what;
+        regex_match(url.c_str(), what, ex);
+        
+        // The web domain 
+        auto const host = what[2];
+        
+        // Set the port type
+        string port = string(what[1].first, what[1].second);
+        if (string(what[1].first, what[1].second)=="https") port = "https"; 
+        else port = "http";
+        auto const target = string(what[4].first, what[4].second);  
+        
+        // Protocol version
+        int version;
+        if (!string(what[6].first, what[6].second).empty() && !std::strcmp("1.0", what[6].second)) version= 11;
+        else version=10;
+		
+        // The io_context is required for all I/O
+        boost::asio::io_context ioc;
+        
+        // The SSL context is required, and holds certificates
+        ssl::context ctx{ssl::context::tls_client};
+        
+        // Verify the remote server's certificate
+        ctx.set_verify_mode(ssl::verify_peer | ssl::context::verify_fail_if_no_peer_cert);    
+        ctx.set_default_verify_paths();
+        
+        // tag::ctx_setup_source[]
+        boost::certify::enable_native_https_server_verification(ctx);
+        // end::ctx_setup_source[]        
+        		        
+        // These objects perform our I/O
+        tcp::resolver resolver{ioc};
+        ssl::stream<tcp::socket> stream{ioc, ctx};        
+        
+        // tag::stream_setup_source[]
+        boost::certify::set_server_hostname(stream, string(what[2].first, what[2].second));
+        boost::certify::sni_hostname(stream, host);
```

Comment:
These are both the same. Is that intended?

Reply:
I'm not sure. See [https://github.com/djarek/certify/blob/master/examples/get_page.cpp](https://github.com/djarek/certify/blob/master/examples/get_page.cpp)